routing/base/astar_algorithm.hpp

```diff
@@ -241,10 +248,11 @@ template <typename TGraph>
 constexpr typename TGraph::TWeightType AStarAlgorithm<TGraph>::kZeroDistance;
 
 template <typename TGraph>
-template <typename VisitVertex, typename AdjustEdgeWeight>
+template <typename VisitVertex, typename AdjustEdgeWeight, typename FilterStates>
 void AStarAlgorithm<TGraph>::PropagateWave(TGraphType & graph, TVertexType const & startVertex,
                                            VisitVertex && visitVertex,
                                            AdjustEdgeWeight && adjustEdgeWeight,
+                                           FilterStates && filterStates,
```

Comment:
We probably should use universal references for all other callbacks too.

Reply:
I think it's hard to benefit from usage of universal references in FindPath/FindPathBidirectional/AdjustRoute.
It's better to pass callback by value once per route calculation than check callback exists every step. And I think current code is more clear.
https://github.com/mapsme/omim/pull/7614#pullrequestreview-78653424